ods_ci/tests/Resources/Page/ModelRegistry/ModelRegistry.resource

```diff
@@ -0,0 +1,400 @@
+    Upload File In The Workbench     filepath=${JUPYTER_NOTEBOOK_FILEPATH}    workbench_title=${WORKBENCH_TITLE}
+    ...         workbench_namespace=${PRJ_TITLE}
+    Download Python Client Dependencies    ${MR_PYTHON_CLIENT_FILES}    ${MR_PYTHON_CLIENT_WHL_VERSION}
+    Upload Python Client Files In The Workbench    ${MR_PYTHON_CLIENT_FILES}
+    Upload Certificate To Jupyter Notebook    ${CERTS_DIRECTORY}/domain.crt
+    ${self_managed} =    Is RHODS Self-Managed
+    IF  ${self_managed}    Upload Certificate To Jupyter Notebook    openshift_ca.crt
+    Jupyter Notebook Can Query Model Registry     ${JUPYTER_NOTEBOOK}
+    SeleniumLibrary.Switch Window    ${handle}
+    Stop Workbench    workbench_title=${WORKBENCH_TITLE}
+    SeleniumLibrary.Close All Browsers
+
+Model Registry Post Upgrade Scenario
+    [Documentation]    Verifies that registered model/version in pre-upgrade is present after the upgrade
+    # If the component is not enabled either it got set to Removed by the upgrade or the pre-upgrade scenario failed/
+    # was not run
+    Set Library Search Order    SeleniumLibrary
+    RHOSi Setup
+    Component Should Be Enabled    modelregistry
+    Launch Dashboard    ${TEST_USER.USERNAME}    ${TEST_USER.PASSWORD}    ${TEST_USER.AUTH_TYPE}
+    ...    ${ODH_DASHBOARD_URL}    ${BROWSER.NAME}    ${BROWSER.OPTIONS}
+    Open Model Registry Dashboard Page
+    SeleniumLibrary.Page Should Contain Element
+    ...    xpath:${MR_TABLE_XPATH}/tbody/tr/td[@data-label="Model name"]//a[.="${MR_REGISTERED_MODEL_NAME}"]
+    SeleniumLibrary.Page Should Contain Element
+    ...    xpath:${MR_TABLE_XPATH}/tbody/tr/td[@data-label="Owner"]//p[.="${MR_REGISTERED_MODEL_AUTHOR}"]
+    SeleniumLibrary.Page Should Contain Element    xpath:${MR_TABLE_XPATH}/tbody/tr/td[@data-label="Labels" and .="-"]
+    SeleniumLibrary.Click Element
+    ...    xpath:${MR_TABLE_XPATH}/tbody/tr/td[@data-label="Model name"]//a[.="${MR_REGISTERED_MODEL_NAME}"]
+    Maybe Wait For Dashboard Loading Spinner Page
+    SeleniumLibrary.Page Should Contain Element
+    ...    xpath:${MR_VERSION_TABLE_XPATH}/tbody/tr/td[@data-label="Version name"]//a[.="${MR_REGISTERED_MODEL_VERSION}"]
+    SeleniumLibrary.Page Should Contain Element
+    ...    xpath:${MR_VERSION_TABLE_XPATH}/tbody/tr/td[@data-label="Author" and .="${MR_REGISTERED_MODEL_AUTHOR}"]
+    SeleniumLibrary.Close All Browsers
+    ${return_code}    ${output}=    Run And Return Rc And Output
+    ...    oc delete project ${PRJ_TITLE} --force --grace-period=0
+    Should Be Equal As Integers	  ${return_code}	 0
+    Log    ${output}
+    ${NAMESPACE_MODEL_REGISTRY}=    Get Model Registry Namespace From DSC
+    Set Suite Variable    ${NAMESPACE_MODEL_REGISTRY}
+    Remove Model Registry
+    Remove Deployment Files    ${CERTS_DIRECTORY}
+    Remove Deployment Files    ${MODELREGISTRY_BASE_FOLDER}/Python_Dependencies
+    Disable Model Registry If Needed
+    RHOSi Teardown
```

Comment:
could you move these lines into a keyword to use as teardown of the post upgrade test? So if any of the previous checks fail, we are sure to clean the env